src/main/scala/mesosphere/marathon/core/task/termination/impl/KillStreamWatcher.scala

```diff
@@ -22,10 +23,16 @@ object KillStreamWatcher extends StrictLogging {
     *
     * See [[mesosphere.marathon.stream.EnrichedSource$.eventBusSource]]
     */
-  private def killedInstanceIds(eventStream: akka.event.EventStream): Source[Instance.Id, Cancellable] = {
+  private def killedInstanceIds(eventStream: akka.event.EventStream, instancesToCheck: Iterable[Instance]): Source[Instance.Id, Cancellable] = {
     // MaxValue causes a dynamically sized buffer to be used.
     val bufferSize = Int.MaxValue
     val overflowStrategy = OverflowStrategy.fail
+
+    val alreadyConsideredTerminal = instancesToCheck
```

Comment:
why was this change needed?

Reply:
I mean how come we did not need this for apps but need it for pods?

Reply:
We didn't test apps use case. We need this to not block and wait 10 minutes for the reconciliation if the instance is already terminal (for example in case of unreachable).